Chapter6/reliable_fsm.erl

```diff
@@ -0,0 +1,120 @@
+-module(reliable_fsm).
+-export([start/0, stop/0, un_hook/0, hook/0, other_un_hook/0, other_hook/0, call/1, get_call/1]).
+-export([idle/0, dial/0, wait_answer/1, ringing/1, connected/1]).
+
+start() -> register(cell, spawn(reliable_fsm, idle, [])).
+stop() 	-> cell ! stop.
+
+un_hook() 			-> msg(off_hook).
+hook() 				-> msg(on_hook).
+other_un_hook() 	-> msg(other_off_hook).
+other_hook() 		-> msg(other_on_hook).
+call(Number) 		-> msg({Number, calling}).
+get_call(Number) 	-> msg({Number, incoming}).
+
+
+% Exercise 6-2: A Reliable Mutex Semaphore
```

Comment:
I've tried to use debugger and other ways of going around the problem but I can't seem to find why linking to one process crashes the second one on wait